src/config.rs

```diff
@@ -120,12 +135,21 @@ pub fn config_path() -> Result<PathBuf> {
 ///
 /// - Fails if the home directory cannot be found
 /// - Fails if the config directory cannot be created
-pub fn save_config(config: &Config) -> Result<()> {
-    let config_path = config_path()?;
-    let mut file = File::create(config_path)?;
-    let config = toml::to_string(config)?;
-    file.write_all(config.as_bytes())?;
-    Ok(())
+
+pub fn save_config(config: &Config) -> Result<(), ConfigError> {
+    // Tries to catch the config path
+    let config_path = config_path().map_err(|_| ConfigError::CreateConfigDir)?;
+    // Tries to create the config file
+    let mut file = File::create(&config_path).map_err(|_| ConfigError::CreateConfigFile)?;
+    // Tries to serialize the config to a string
+    let config = toml::to_string(config).map_err(|_| ConfigError::SaveConfigFile)?;
+    // Tries to write the config to the file
+    match file.write_all(config.as_bytes()) {
```

Comment:
Also return a `std::io::Error`, so we might have a conflict when converting from these errors with `thiserror` crate.
Don't know how to solve, but I think you'll figure something out.